horovod/run/run.py

```diff
@@ -0,0 +1,472 @@
+        if verbose:
+            print("Host-to-host interface checking successful.")
+        # Determine a set of common interfaces for task-to-task communication.
+        common_intfs = set(driver.task_addresses_for_tasks(0).keys())
+        for index in range(1, num_hosts):
+            common_intfs.intersection_update(
+                driver.task_addresses_for_tasks(index).keys())
+        if not common_intfs:
+            raise Exception(
+                'Unable to find a set of common task-to-task communication '
+                'interfaces: %s'
+                % [(index, driver.task_addresses_for_tasks(index))
+                   for index in range(num_hosts)])
+        return common_intfs
+    finally:
+        driver.shutdown()
+
+
+def _is_open_mpi_installed():
+    output = six.StringIO()
+    command = 'mpirun --version'
+    try:
+        exit_code = safe_shell_exec.execute(command, stdout=output,
+                                            stderr=output)
+        output_msg = output.getvalue()
+    except Exception:
+        print(traceback.format_exc(), file=sys.stderr)
+        return False
+    finally:
+        output.close()
+
+    if exit_code == 0:
+        if 'Open MPI' not in output_msg:
+            print('Open MPI not found in output of mpirun --version.',
+                  file=sys.stderr)
+            return False
+        else:
+            return True
+    else:
+        print("Was not able to run %s:\n%s" % (command, output_msg),
+              file=sys.stderr)
+        return False
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Horovod Runner')
+
+    parser.add_argument('-v', '--version', action="store_true", dest="version",
+                        help="Show Horovod version.")
+
+    parser.add_argument('-np', '--num-proc', action="store", dest="np",
+                        type=int,
+                        help="Number of processes which should be equal to "
+                             "the total number of available GPUs.")
+
+    parser.add_argument('-p', '--ssh-port', action="store", dest="ssh_port",
+                        type=int,
+                        help="SSH port on all the workers.")
+
+    parser.add_argument('-H', '--host', action="store", dest="host",
+                        help="To specify the list of hosts on which "
+                             "to invoke processes. Takes a comma-delimited "
+                             "list of hosts.")
+
+    parser.add_argument('--disable-cache', action="store_true",
+                        dest="disable_cache",
+                        help="If flag is set, horovod run will not cache the "
+                             "initial checks and execute them every time.")
+    parser.add_argument('--horovod-start-timeout', action="store",
+                        dest="start_timeout",
+                        help="Horovodrun has to perform all the checks and and "
```

Comment:
Typo: "and and".

What is the default?

Reply:
it is 600 seconds. It can also be specified using env variable `HOROVOD_START_TIMEOUT`

Reply:
Let's add to docs